adapters/tradplus/tradplus.go

```diff
@@ -0,0 +1,176 @@
+package tradplus
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/macros"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint *template.Template
+}
+
+// Builder builds a new instance of the tradplus adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	return &adapter{
+		endpoint: template,
+	}, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	adapterRequest, errs := a.makeRequest(request)
+	if errs != nil {
+		return nil, errs
+	}
+	return []*adapters.RequestData{adapterRequest}, nil
+}
+
+func (a *adapter) makeRequest(request *openrtb2.BidRequest) (*adapters.RequestData, []error) {
+
+	tradplusExt, err := getImpressionExt(&request.Imp[0])
```

Comment:
Please link to your prebid.github.io PR and make it clear in that PR that the TradPlus only looks at the ext for the first impression and ignored the rest. Some other adapters do it, but its not a publisher expected behavior.

Reply:
ok，I've added to the first comment if anything else is needed ?

Reply:
Yes. New adapters must also open a PR against the Prebid Docs repository here: https://github.com/prebid/prebid.github.io/

Please see the [User Documentation](https://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html#user-documentation) section of our new adapter developer guide.

Reply:
Follow up question: adapter code takes `imp[0].ext` and uses it to build endpoint url. Then sets` imp[0].ext` to nil, L49: `request.Imp[0].Ext = nil`
Other impressions are passed as is, with imp[n].ext. 
Is this expected? Do you want to set every imp[n].ext to nil?